servlet.py

```diff
@@ -121,6 +124,49 @@ class BaseHandler(tornado.web.RequestHandler):
     def initialize(self):
         self.callback = self.get_argument('callback', default=None)
 
+    @classmethod
+    def initPairsGraph(cls):
```

Comment:
@unhammer Because I thought the graph of pairs should belong to the class rather than an instance of it, as all instances will access the same graph. It is also sufficient to initialize the graph and find all the shortest pairs only once this way, not duplicating the processing for each instance.